File: benchmarks/csharp/ForLoop.cs
// Similar to GDScript for_loop benchmarks, but using C#

public partial class ForLoop : BenchmarkCS
{
    private const int ITERATIONS = 1_000_000;

    private int number = 0;

    public void BenchmarkLoopAdd()
    {
        for(int i = 0; i < ITERATIONS; i++)
        {
            number += 1;
        }
    }

    public void BenchmarkLoopCall()
    {
        for(int i = 0; i < ITERATIONS; i++)
        {
            Function();
        }
    }

    public virtual void Function(){}
}

File: gdextension/src/benchmarks/forloop.cpp
#include "forloop.h"

#include <godot_cpp/core/class_db.hpp>

using namespace godot;

void CPPBenchmarkForLoop::_bind_methods() {
	ClassDB::bind_method(D_METHOD("benchmark_loop_add"), &CPPBenchmarkForLoop::benchmark_loop_add);
	ClassDB::bind_method(D_METHOD("benchmark_loop_call"), &CPPBenchmarkForLoop::benchmark_loop_call);
}

void CPPBenchmarkForLoop::function() {}

void CPPBenchmarkForLoop::benchmark_loop_add() {
	int number = 0;
	for (int i = 0; i < iterations; i++)
		number++;
}

void CPPBenchmarkForLoop::benchmark_loop_call() {
	for (int i = 0; i < iterations; i++)
		function();
}

CPPBenchmarkForLoop::CPPBenchmarkForLoop() {}

CPPBenchmarkForLoop::~CPPBenchmarkForLoop() {}
